### ISSUE TITLE
python3Packages.mysql-connector fails to install on macOS

### ISSUE THREAD

**Issue by neosimsim** [NixOS/nixpkgs#130376]
**Describe the bug**
I try to setup python with [mysql-connector](https://search.nixos.org/packages?channel=21.05&show=python38Packages.mysql-connector&from=0&size=50&sort=relevance&query=mysql-connector). The build fails with
```
installing
Executing pipInstallPhase
/private/tmp/nix-build-python3.8-mysql-connector-8.0.24.drv-1/source/dist /private/tmp/nix-build-python3.8-mysql-connector-8.0.24.drv-1/source
ERROR: mysql_connector_python-8.0.24-cp38-cp38-macosx_11_0_x86_64.whl is not a supported wheel on this platform.
builder for '/nix/store/1icjz2rid3nxhi4qr5l3cl19k11mfnp2-python3.8-mysql-connector-8.0.24.drv' failed with exit code 1
error: build of '/nix/store/1icjz2rid3nxhi4qr5l3cl19k11mfnp2-python3.8-mysql-connector-8.0.24.drv' failed
```

**To Reproduce**
Run
```
nix-build -A python3Packages.mysql-connector
```

I'm not sure if this is an issue in `nixpkgs` or `mysql-connector` but installing `mysql-connector` with `pip` works fine.

**Comment by neosimsim** [comment]
macOS version:
```
% sw_vers 
ProductName:    macOS
ProductVersion: 11.1
BuildVersion:   20C69
```

**Comment by neosimsim** [comment]
@turion and I tracked the issue down to https://github.com/mysql/mysql-connector-python/commit/d1e89fd3d7391084cdf35b0806cb5d2a4b413654. I suggest to add a patch to the nixpkgs derivation, which reverses the commit.

**Comment by neosimsim** [comment]
Resolved by #130734
